raiden/network/rpc/client.py

```diff
@@ -1112,13 +1114,22 @@ def deploy_single_contract(
         transaction_mined = self.poll_transaction(transaction_sent)
         contract_address = to_canonical_address(transaction_mined.receipt["contractAddress"])
 
+        if not was_transaction_successfully_mined(transaction_mined):
```

Comment:
While looking at the implementation, I saw that `was_transaction_successfully_mined`, according to the type annotation returns `Optional[Dict[str, Any]]`. Can you fix that and do you have any idea why mypy didn't catch that?